py-polars/tests/unit/io/test_csv.py

```diff
@@ -1737,3 +1737,17 @@ def test_invalid_csv_raise() -> None:
     "SK0127960V000","SK BT 0018977","
     """.strip()
         )
+
+
+@pytest.mark.slow()
+def test_partial_read_compressed_web_file() -> None:
+    url = "https://raw.githubusercontent.com/hadley/mastering-shiny/main/neiss/injuries.tsv.gz"
```

Comment:
The test suite should not be reliant on external content if at all possible. Please include a small example (e.g. a multiline string) and read from that.

Reply:
This issue only triggers when reading a compressed file with a decompressed size of more than 8192 bytes.

Reply:
> This issue only triggers when reading a compressed file with a decompressed size of more than 8192 bytes.

There's a trick to rapidly generate large test frames, which you can use to create sufficient CSV data for the test, like so:
```python
df = pl.DataFrame({
  "idx": range(2_500_000),
  "dt" : date(2025,12,31), 
  "txt": "hello world",
})
csv_data = df.write_csv()
```
This only takes ~0.175 secs on my laptop and creates uncompressed data of ~60MB. You could then gz-compress it to run through the validating code 👌

Reply:
I like that approach! Just implemented it in the test.